python/pyspark/ml/classification.py

```diff
@@ -2290,10 +2291,15 @@ def featureImportances(self) -> Vector:
         """
         return self._call_java("featureImportances")
 
-    @property
+    @cached_property
```

Comment:
cache the result to avoid re-generating and re-caching the submodels (the trees) in the server side.